backends/memcache.go

```diff
@@ -67,44 +69,108 @@ func (m *Memcache) Execute(cmd string, args ...interface{}) (interface{}, error)
 	return "inside GEt", nil
 }
 
-func (m *Memcache) Delete(keys ...interface{}) int {
-	return 1
+func (m *Memcache) Delete(keys ...interface{}) (int, error) {
```

Comment:
I wanted to handle all the errors in this module itself. With these changes errors should now be handled in clients as well. Was there any use case where in you had to return errors?.

There can one buffered-error-channel where in you can keep pushing errors with client-id or something. So if you are interested in errors you can subscribe to that channel.

Let me know your thoughts.